tensor2tensor/data_generators/generator_utils.py

```diff
@@ -338,13 +338,19 @@ def generate():
 
         # Use Tokenizer to count the word occurrences.
         with tf.gfile.GFile(filepath, mode="r") as source_file:
-          file_byte_budget = 3.5e5 if filepath.endswith("en") else 7e5
+          file_byte_budget = 1e6 if filepath.endswith("en") else 1e6
+          counter = 0
+          countermax = int(source_file.size() / 1e6)
           for line in source_file:
-            if file_byte_budget <= 0:
-              break
-            line = line.strip()
-            file_byte_budget -= len(line)
-            yield line
+            if counter < countermax:
+              counter += 1
+            else:
+              if file_byte_budget <= 0:
+                break
+              line = line.strip()
+              file_byte_budget -= len(line)
+              counter = 0
+              yield line
```

Comment:
* Maybe it would be better to have this file_byte_budget as a separate PR as it is not really related to wmt.py refactoring and (unlike the refactoring) it changes the behavior and may affect BLEU.
* Now I realized that even better than solution would be to increase file_byte_budget dynamically as long as the subword/BPE algorithm ends up with min_count=1 (or 2 or a user-specified constant). Or (to make it faster) we could make sure the space-delimited-tokens (or e.g. 10-bytes-tokens for non-space languages) vocabulary size is significantly bigger than  the expected subword vocab size.

Reply:
Let's not make it too complicated. Using a parameter `file_budget` (per file) should be good enough for great most cases, no?